### PR TITLE
Add docs for YAML configuration files

### PR DESCRIPTION
**Pull request by BoloniniD**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add documentation for YAML configuration.

### CODE REVIEW

**Comment by BoloniniD** [comment]
Created a new PR for  #21858 documentation.

File: docs/en/operations/configuration-files.md [hunk 1]
```diff
@@ -55,6 +55,68 @@ $ cat /etc/clickhouse-server/users.d/alice.xml
 </yandex>
 ```
 
+## YAML examples {#example}
```
Comment (alexey-milovidov) [inline docs/en/operations/configuration-files.md:58]: Let's add a link to example in the repository (default config in YAML with comments).

File: docs/en/operations/configuration-files.md [hunk 2]
```diff
@@ -5,9 +5,9 @@ toc_title: Configuration Files
 
 # Configuration Files {#configuration_files}
 
-ClickHouse supports multi-file configuration management. The main server configuration file is `/etc/clickhouse-server/config.xml`. Other files must be in the `/etc/clickhouse-server/config.d` directory.
+ClickHouse supports multi-file configuration management. The main server configuration file is `/etc/clickhouse-server/config.xml` or `/etc/clickhouse-server/config.yaml`. Other files must be in the `/etc/clickhouse-server/config.d` directory. Note, that any configuration file can be written either in XML or YAML, but mixing formats in one file is not supported.
 
-All the configuration files should be in XML format. Also, they should have the same root element, usually `<yandex>`.
+All the configuration files should be in XML or YAML formats. All XML files should have the same root element, usually `<yandex>`. As for YAML, writing `yandex:` is not needed, parser will insert it automatically.
```
Comment (alexey-milovidov) [inline docs/en/operations/configuration-files.md:10]: Not just "not needed" but should not be present.

File: docs/en/operations/configuration-files.md [hunk 3]
```diff
@@ -5,9 +5,9 @@ toc_title: Configuration Files
 
 # Configuration Files {#configuration_files}
 
-ClickHouse supports multi-file configuration management. The main server configuration file is `/etc/clickhouse-server/config.xml`. Other files must be in the `/etc/clickhouse-server/config.d` directory.
+ClickHouse supports multi-file configuration management. The main server configuration file is `/etc/clickhouse-server/config.xml` or `/etc/clickhouse-server/config.yaml`. Other files must be in the `/etc/clickhouse-server/config.d` directory. Note, that any configuration file can be written either in XML or YAML, but mixing formats in one file is not supported.
```
Comment (alexey-milovidov) [inline docs/en/operations/configuration-files.md:8]: Add:

For example, you can have main configs as `config.xml` and `users.xml` and write additional files in `config.d` and `users.d` directories in `.yaml`.

File: programs/server/config-example.yaml [hunk 4]
```diff
@@ -0,0 +1,946 @@
...
+        disableProtocols: 'sslv2,sslv3'
+        preferServerCiphers: true
+
+        # Use for self-signed: verificationMode: none
+        invalidCertificateHandler:
+            # Use for self-signed: name: AcceptCertificateHandler
+            name: RejectCertificateHandler
+
+# Default root page on http[s] server. For example load UI from https://tabix.io/ when opening http://localhost:8123
+# <http_server_default_response><![CDATA[<html ng-app="SMI2"><head><base href="http://ui.tabix.io/"></head><body><div ui-view="" class="content-ui"></div><script src="http://loader.tabix.io/master.js"></script></body></html>]]></http_server_default_response>
```
Comment (alexey-milovidov) [inline programs/server/config-example.yaml:202]: This has to be adapted.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**